### PR TITLE
Covid-19 page draft implementation

### PR DESCRIPTION
**Pull request by devonbush**
Adds feature flags for a new covid19 page, and for allowing us to release advanced search while still hiding the facet controls.  This also brings back capability for the 'most popular' and 'most recent' buttons

### CODE REVIEW

File: app/javascript/lib/scp-api.js [hunk 1]
```diff
@@ -167,9 +167,12 @@ export async function fetchDownloadSize(accessions, fileTypes, mock=false) {
...
  *
  * @param {String} type Type of query to perform (study- or cell-based)
- * @param {String} terms User-supplied query string
- * @param {Object} facets User-supplied list facets and filters
- * @param {Integer} page User-supplied list facets and filters
+ * @param {Object} searchParams  User-supplied search parameters including
+ *            {String}  terms: User-supplied query string
+ *            {Object}  facets: User-supplied list facets and filters
+ *            {Integer} page: User-supplied list facets and filters
+ *            {String}  order: User-supplied query ordering field
```
Comment (eweitz) [inline app/javascript/lib/scp-api.js]: Slightly clearer:

```suggestion
 *            {String}  order: User-supplied results ordering field
```

File: app/javascript/packs/application.js [hunk 2]
```diff
@@ -50,6 +51,11 @@ document.addEventListener('DOMContentLoaded', () => {
       logClick(event)
     })
   }
+  if (document.getElementById('covid19-page-content')) {
```
Comment (eweitz) [inline app/javascript/packs/application.js]: Would it help to track user flows through the COVID-19 advanced search UI?

Applying `logPageView` and `logClick` here would enable that.  Our Google Analytics fallback can provide basic event flows right now, while Mixpanel progresses upstream.

File: app/views/site/covid19.html.erb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    }
+  </style>
+  <div id="main-banner-logo">
+    <p id='main-banner-tag-line' class="scp-banner-position"></p>
+  </div>
+</div>
+<div class="container-fluid home-page-fix" id="wrap">
+  <div class="row section-pad" id="main-body">
+    <div class="col-md-12">
+      <h4> COVID-19 Studies</h4>
```
Comment (eweitz) [inline app/views/site/covid19.html.erb]:
```suggestion
      <h4>COVID-19 Studies</h4>
```

File: app/views/site/index.html.erb [hunk 4]
```diff
@@ -31,6 +37,8 @@
     </div>
   </div>
 </div>
+<%# render feature flags to html so they can be found by react components #%>
+<input type="hidden" id="feature-flags" value='<%= User.feature_flags_for_user(current_user).to_json %>'/>
 
 <script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
     $(window).scroll(bindScroll);
```
Comment (eweitz) [inline app/views/site/index.html.erb]: See main comment in `feature-flags.js`.

```suggestion
<script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
    window.SCP.featureFlags = '<%= User.feature_flags_for_user(current_user).to_json %>';
    $(window).scroll(bindScroll);
```

File: app/views/site/covid19.html.erb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    </div>
+  </div>
+</div>
+<input type="hidden" id="feature-flags" value='<%= User.feature_flags_for_user(current_user).to_json %>'/>
+<script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
+    $(document).ready(function() {
+        var main = $('.home-page-fix');
+        var height = main.actual('height');
+        main.css('min-height', height - 245);
+    });
```
Comment (eweitz) [inline app/views/site/covid19.html.erb]: See main comment in `feature-flags.js`.

```suggestion
<script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
    window.SCP.featureFlags = '<%= User.feature_flags_for_user(current_user).to_json %>';

    $(document).ready(function() {
        var main = $('.home-page-fix');
        var height = main.actual('height');
        main.css('min-height', height - 245);
    });
```

File: app/javascript/lib/feature-flags.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export function getFlagValue(flagId) {
+  return getAllFlags()[flagId]
+}
+
+export function getAllFlags() {
+  // for now, read it off the home page.  Eventually, this will want to be an API call
+  return JSON.parse($('#feature-flags').attr('value'))
```
Comment (eweitz) [inline app/javascript/lib/feature-flags.js]: See related suggestions in `covid19.html.erb` and `index.html.erb`.  Benefits of making Rails variables available to JavaScript in our React code via `window.SCP`:

* Enables easier automated testing via Jest
* Eliminates reliance on jQuery and DOM parsing generally
* More consistent with general pattern for this in our React codebase, e.g. `UserProvider.js`
* Faster and simpler to refactor when we use Node environment to get Rails execution context (SCP-2237)

```suggestion
  return ('SCP' in window) ? window.SCP.featureFlags : {test: true}
```

File: app/javascript/components/covid19/Covid19PageContent.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import ResultsPanel from 'components/ResultsPanel'
+import StudySearchProvider from 'components/search/StudySearchProvider'
+import SearchFacetProvider from 'components/search/SearchFacetProvider'
+import UserProvider from 'components/UserProvider'
+import ErrorBoundary from 'lib/ErrorBoundary'
+
+/**
+ * Wrapper component for search and result panels
+ */
+export default function HomePageContent() {
```
Comment (eweitz) [inline app/javascript/components/covid19/Covid19PageContent.js]:
```suggestion
export default function Covid19PageContent() {
```

**Comment by bistline** [comment]
The failing build I think is due to some possibly faulty logic in either the test for `SummaryStatsUtils`, or possibly an actual bug in that code.  But it does not impact this functionality so should not be a blocker for merge.

**Pull request merged**